daemon/nvidia_linux.go

```diff
@@ -30,19 +33,28 @@ var allNvidiaCaps = map[nvidia.Capability]struct{}{
 }
 
 func init() {
-	if _, err := exec.LookPath(nvidiaHook); err != nil {
-		// do not register Nvidia driver if helper binary is not present.
-		return
-	}
-	capset := capabilities.Set{"gpu": struct{}{}, "nvidia": struct{}{}}
-	nvidiaDriver := &deviceDriver{
-		capset:     capset,
-		updateSpec: setNvidiaGPUs,
-	}
-	for c := range allNvidiaCaps {
-		nvidiaDriver.capset[string(c)] = struct{}{}
+	// Register Nvidia driver if Nvidia helper binary is present.
+	// Else, register AMD driver if AMD helper binary is present.
+	if _, err := exec.LookPath(nvidiaHook); err == nil {
+		capset := capabilities.Set{"gpu": struct{}{}, "nvidia": struct{}{}}
```

Comment:
Also curious if `gpu` and `nvidia` should be added to `allNvidiaCaps`, so that no merging is needed 🤔 (not sure)

Reply:
Let's keep that out of scope for this PR. I think the `allNvidiaCaps` is supposed to mirror our `NVIDIA_DRIVER_CAPABILITIES`. These are different from the Docker capabilities (which include `gpu` and `nvidia`). I will revisit this when I start working on a `--gpus` to CDI migration plan in the next couple of weeks.